src/client/datascience/errors/errorHandler.ts

```diff
@@ -256,16 +289,62 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
             if ((purpose === 'restart' || purpose === 'start') && handler) {
                 await handler(err, defaultErrorMessage);
             } else {
+                void this.displayErrorsInCell(defaultErrorMessage, cellToDisplayErrors);
                 this.applicationShell.showErrorMessage(defaultErrorMessage).then(noop, noop);
             }
         } else {
             // Some errors have localized and/or formatted error messages.
-            this.applicationShell
-                .showErrorMessage(getCombinedErrorMessage(errorPrefix, err.message || err.toString()))
-                .then(noop, noop);
+            const message = getCombinedErrorMessage(errorPrefix, err.message || err.toString());
+            void this.displayErrorsInCell(message, cellToDisplayErrors);
+            this.applicationShell.showErrorMessage(message).then(noop, noop);
         }
         traceError('DataScience Error', err);
     }
+    private async displayErrorsInCell(errorMessage: string, cellToDisplayErrors?: NotebookCell) {
+        if (!cellToDisplayErrors || !errorMessage) {
+            return;
+        }
+        const associatedkernel = this.serviceContainer
+            .get<IKernelProvider>(IKernelProvider)
+            .get(cellToDisplayErrors.notebook);
+        if (!associatedkernel) {
+            return;
+        }
+        // Sometimes the cells are still running, wait for 1s for cells to finish & get cleared,
+        // Then display the error in the cell.
+        const stopWatch = new StopWatch();
+        while (stopWatch.elapsedTime <= 1_000 && associatedkernel.hasPendingCells) {
+            await sleep(100);
```

Comment:
Would it make sense to interrupt the current execution instead?

Reply:
No, it will stop. The existing code will still the execution.
The problem is this code gets called just when kernel dies, hence the cells are stored after that, within a few Ms.
Hence the max delay of 1s.

I wish the api allowed us to update cell output without all of this api. But these are a few exceptions